x-pack/plugins/security_solution/public/resolver/view/panel.tsx

```diff
@@ -3,191 +3,283 @@
-export const Panel = memo(function Event({ className }: { className?: string }) {
-  interface ProcessTableView {
-    name: string;
-    timestamp?: Date;
-    event: ResolverEvent;
-  }
+/**
+ * The team decided to use this table to determine which breadcrumbs/view to display:
+ *
+ * | Crumb/Table            | &crumbId                   | &crumbEvent              |
+ * | :--------------------- | :------------------------- | :----------------------  |
+ * | all processes/default  | null                       | null                     |
+ * | process detail         | entity_id of process       | null                     |
+ * | relateds count by type | entity_id of process       | 'all'                    |
+ * | relateds list 1 type   | entity_id of process       | valid related event type |
+ * | related event detail   | event_id of related event  | entity_id of process     |
+ *
+ * This component implements the strategy laid out above by determining the "right" view and doing some other housekeeping e.g. effects to keep the UI-selected node in line with what's indicated by the URL parameters.
+ *
+ * @returns {JSX.Element} The "right" table content to show based on the query params as described above
+ */
+const PanelContent = memo(function PanelContent() {
+  const history = useHistory();
+  const urlSearch = history.location.search;
+  const dispatch = useResolverDispatch();
 
-  const { processNodePositions } = useSelector(selectors.processNodePositionsAndEdgeLineSegments);
-  const { timestamp } = useContext(SideEffectContext);
+  const queryParams: CrumbInfo = useMemo(() => {
+    return { crumbId: '', crumbEvent: '', ...querystring.parse(urlSearch.slice(1)) };
+  }, [urlSearch]);
 
-  const processTableView: ProcessTableView[] = useMemo(
-    () =>
-      [...processNodePositions.keys()].map((processEvent) => {
-        let dateTime;
-        const eventTime = event.eventTimestamp(processEvent);
-        const name = event.eventName(processEvent);
-        if (eventTime) {
-          const date = new Date(eventTime);
-          if (isFinite(date.getTime())) {
-            dateTime = date;
-          }
-        }
-        return {
-          name,
-          timestamp: dateTime,
-          event: processEvent,
-        };
-      }),
-    [processNodePositions]
-  );
+  const graphableProcesses = useSelector(selectors.graphableProcesses);
+  const graphableProcessEntityIds = useMemo(() => {
+    return new Set(graphableProcesses.map(event.entityId));
+  }, [graphableProcesses]);
+  // The entity id in query params of a graphable process (or false if none is found)
+  // For 1 case (the related detail, see below), the process id will be in crumbEvent instead of crumbId
+  const idFromParams = useMemo(() => {
+    if (graphableProcessEntityIds.has(queryParams.crumbId)) {
+      return queryParams.crumbId;
+    }
+    if (graphableProcessEntityIds.has(queryParams.crumbEvent)) {
+      return queryParams.crumbEvent;
+    }
+    return '';
```

Comment:
This return value (`''`) is confusing to me. What is the meaning of an `id` that is an empty string? If there is no id found, why not return undefined or null?